### ISSUE TITLE
`Won't Fix` alerts are still shown

### ISSUE THREAD

**Issue by laurentsimon** [github/codeql-action#986]
I'm one of the maintainers of the [scorecard's project](github.com/ossf/scorecard) and we integrated with the code scanning a few months ago.

One user https://github.com/ossf/scorecard-action/issues/143 reported that the results keeps showing after the file is updated via dependabot, even though the results were marked as `Won't Fix`

TL;DR: scorecard creates alerts if the actions defined in a GitHub workflows are not pinned by hash ( https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions). Once in a while, dependabot creates PR to update the hash when a new version of an action becomes available. After accepting a dependabot PR, the code scanning shows previous alerts that were tagged with `Won't Fix`.

My feature request is that the code scanning dashboard should only show alerts for code that has been changed "meaningfully" by developers, rather than simply updated via dependabot and renovatebot for version updates.

**Comment by laurentsimon** [comment]
yes this is exactly the behavior. Version updates don't change the code semantic, so one way to improve would simply be to look at previous commit and see if the line was changed by a dependabot/renovatebot PR only. If that's the case, it'd be safe to assume the code has not changed.

Would this be practical? Would you anticipate problems with this approach?

**Comment by azeemshaikh38** [comment]
A suggestion - could `github/codeql-action/upload-sarif` action be expanded to have a `exclude` option, wherein users can specify `ruleIDs` they want to ignore? That enables users to completely disable rules/checks they are not interested in.

**Comment by azeemshaikh38** [comment]
Or better yet - the UI should have an option to allow users to ignore all failures related to a `ruleID`.

**Comment by laurentsimon** [comment]
cc @josepalafox

**Comment by josepalafox** [comment]
Hi @laurentsimon I talked with a few folks and received the following overview and suggestion:

 To provide some technical details here, Code Scanning uses the [SARIF `partialFingerprints` property](https://docs.oasis-open.org/sarif/sarif/v2.1.0/csprd01/sarif-v2.1.0-csprd01.html#_Ref507591746) to determine whether an alert is the same as a previously seen alert. (Specifically it uses the fingerprint called `primaryLocationLineHash`.)

When the CodeQL Action is used to upload SARIF files we check if this property is present, and if it is not we calculate a value based on the hash of the surrounding lines in the file. If you make changes to the line containing the alert, the partial fingerprint will change.

If a tool wishes to override this definition of if an alert is "new" or not, they can do this by providing their own value for the `partialFingerprints.primaryLocationLineHash` which remains stable when certain properties of the line change.

**Comment by laurentsimon** [comment]
Thank you @josepalafox !

I will take a look.
